kayttooikeus-service/src/main/java/fi/vm/sade/kayttooikeus/service/impl/OrganisaatioHenkiloServiceImpl.java

```diff
@@ -227,12 +217,54 @@ private OrganisaatioHenkilo findFirstMatching(OrganisaatioHenkiloUpdateDto organ
                 + organisaatioHenkilo.getOrganisaatioOid()));
     }
 
+    // passivoi organisaatiohenkilön ja poistaa käyttöoikeudet
+    private void passivoiOrganisaatioHenkiloJaPoistaKayttooikeudet(OrganisaatioHenkilo organisaatioHenkilo, Henkilo kasittelija, String selite) {
+        organisaatioHenkilo.setPassivoitu(true);
+        Set<KayttoOikeusRyhmaTapahtumaHistoria> historia = organisaatioHenkilo.getMyonnettyKayttoOikeusRyhmas().stream()
+                .map(myonnettyKayttoOikeusRyhmaTapahtuma -> myonnettyKayttoOikeusRyhmaTapahtuma
+                        .toHistoria(kasittelija, KayttoOikeudenTila.SULJETTU, LocalDateTime.now(), selite))
+                .collect(toSet());
+        organisaatioHenkilo.setKayttoOikeusRyhmaHistorias(historia);
+        this.kayttoOikeusRyhmaTapahtumaHistoriaDataRepository.saveAll(historia);
+        this.myonnettyKayttoOikeusRyhmaTapahtumaRepository.deleteAll(organisaatioHenkilo.getMyonnettyKayttoOikeusRyhmas());
+        organisaatioHenkilo.setMyonnettyKayttoOikeusRyhmas(Sets.newHashSet());
+        ldapSynchronizationService.updateHenkiloAsap(organisaatioHenkilo.getHenkilo().getOidHenkilo());
+    }
+
     @Transactional
     @Override
-    public void passivoiOrganisaationHenkilot() {
-        List<String> passiivisetOids = organisaatioClient.getLakkautetutOids();
-        List<OrganisaatioHenkilo> organisaatioHenkilosInLakkautetutOrganisaatios = this.organisaatioHenkiloRepository.findByOrganisaatioOidIn(passiivisetOids);
-        organisaatioHenkilosInLakkautetutOrganisaatios.forEach(oh -> this.passivoiHenkiloOrganisation(oh.getHenkilo().getOidHenkilo(), oh.getOrganisaatioOid()));
+    public void kasitteleOrganisaatioidenLakkautus(String kasittelijaOid) {
```

Comment:
Tämän tekemistä isoista in queryistä voisi varoitella ettei tätä kutsuta ihan huvikseen.

Reply:
Jooh. Untuvalla tämän läpikäynti kesti muutaman minuutin ensimmäisellä kerralla. Mutta kunhan tuo on ajettu kerran, niin tuskinpa noin isoja enää tulee...